hack/generator/pkg/astmodel/struct_definition.go

```diff
@@ -56,6 +56,18 @@ func (definition *StructDefinition) WithIsStorageVersion(isStorageVersion bool)
 	return &result
 }
 
+// RequiredImports returns a list of packages required by this type
+func (definition *StructDefinition) RequiredImports() []*PackageReference {
+	typeImports := definition.StructType.RequiredImports()
+
+	if definition.isResource {
+		metav1Import := NewPackageReference("k8s.io/apimachinery/pkg/apis/meta/v1")
```

Comment:
Should fix the usages of metav1 in here to confirm to the pattern with codeGenContext